File: task_equivalents/llm_translations/llama/optimized/r/statistical_modelling_analysis/02_logistic_regression_odds_ratios.R
library(readr)
library(broom)
library(MASS)

Weekly <- read_csv('data/csv/Weekly.csv')

x01 <- cbind(1, Weekly[, 3:8])
y01 <- ifelse(Weekly$Direction == 'Up', 1, 0)

glm0_fit <- glm(y01 ~ x01 - 1, family = binomial)
print(tidy(glm0_fit))
print(glance(glm0_fit))

prob <- predict(glm0_fit, type = 'response')
pred <- ifelse(prob > 0.5, 1, 0)
print(table(y01, pred))
print(mean(y01 == pred))

File: task_equivalents/llm_translations/llama/optimized/python/statistical_modelling_analysis/02_logistic_regression_odds_ratios.py
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix

# Load data
weekly = pd.read_csv('data/csv/Weekly.csv', index_col=0)

# Prepare data
x01 = weekly.iloc[:, 2:8]
y01 = (weekly['Direction'] == 'Up').astype(int)

# Fit logistic regression model
glm0_fit = LogisticRegression(max_iter=1000)
glm0_fit.fit(x01, y01)

# Print model summary (coefficients)
print("Coefficients:")
print(glm0_fit.coef_)
print("Intercept:")
print(glm0_fit.intercept_)

# Make predictions
prob = glm0_fit.predict_proba(x01)[:, 1]
pred = (prob > 0.5).astype(int)

# Evaluate model
print("\nConfusion Matrix:")
print(confusion_matrix(y01, pred))
print("\nAccuracy:")
print(accuracy_score(y01, pred))